canonical-web-and-design/maas-ui
# Move selected machine state from Machines component into Redux.

## Done
- Moved selected machine state from Machines component into Redux.
- I started writing some action form components and found it a bit unwieldly passing selectedMachines from Machines to about 10 different places at 3-4 levels deep. I think now's the time for it to live in Redux.
- Rewrote "selected machines" tests to assert Redux-y things and moved them from Machines into MachineList

## QA
- Check that selecting and unselecting machines still works as before:
  - Machine checkbox check/uncheck
  - Group checkbox check/uncheck
  - All checkbox check/uncheck
  - Group string e.g. "2 machines, 1 selected"
  - Take action menu enables and can filter to only the machines that can perform action

Fixes #879
